src/main/java/io/cdap/delta/bigquery/MultiGCSWriter.java

```diff
@@ -103,6 +102,20 @@ public synchronized void write(Sequenced<DMLEvent> sequencedEvent) {
     }
   }
 
+  private boolean isJsonFormat(StructuredRecord record) {
+    List<Schema.Field> fields = record.getSchema().getFields();
+    if (fields == null) {
+      return false;
+    }
+    for (Schema.Field field : fields) {
+      Schema.LogicalType logicalType = field.getSchema().getNonNullable().getLogicalType();
```

Comment:
This may be null for fields that are not nullable. Typically I see checks like `isNullable ? field.getSchema().getNonNullable().getType() : field.getSchema().getType()`. https://github.com/cdapio/cdap/blob/aa90c10ed4a7e0638f5d57db52b8d8e41830738c/cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java#L574

Reply:
fixed.